### PR TITLE
Support parquet bloom filter length

### PR DESCRIPTION
**Pull request by letian-jiang**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-rs/issues/4398.

# Rationale for this change
 

# What changes are included in this PR?

If `bloom_filter_length` is available in `ColumnMetaData`, use a single IO to fetch both header and bitset. 

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/src/format.rs [hunk 1]
```diff
@@ -772,25 +773,12 @@ impl TSerializable for Statistics {
   }
 }
 
-impl Default for Statistics {
-  fn default() -> Self {
-    Statistics{
-      max: Some(Vec::new()),
```
Comment (tustvold) [inline parquet/src/format.rs:778]: It occurs to me that this is technically a breaking change

Comment (letian-jiang) [reply parquet/src/format.rs:778]: actually this is replaced by `#[derive(Default)]` at line 641

Comment (letian-jiang) [reply parquet/src/format.rs:778]: if I didn't misunderstand it

File: parquet/src/bloom_filter/mod.rs [hunk 2]
```diff
@@ -289,11 +293,12 @@ impl Sbbf {
                 // this match exists to future proof the singleton hash enum
             }
         }
-        // length in bytes
-        let length: usize = header.num_bytes.try_into().map_err(|_| {
-            ParquetError::General("Bloom filter length is invalid".to_string())
-        })?;
```
[the remark this answers is not in the document]
Comment (letian-jiang) [reply parquet/src/bloom_filter/mod.rs:295]: fixed

File: parquet/src/format.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/format.rs:778]: But that will default the fields to None

**Review by tustvold — APPROVED** [review]
Looks good to me, will merge once CI is green

File: parquet/src/format.rs (hunk 1 above)
Comment (letian-jiang) [reply parquet/src/format.rs:778]: You are right. It is a breaking change.

**Comment by mapleFU** [comment]
Since 2.10 is not released,  can bloom filter length be used now? 🤔

**Comment by tustvold** [comment]
I hadn't realised it wasn't released, but it's a fairly innocuous change so I'm inclined to think this is fine

**Comment by mapleFU** [comment]
Nice! This is helpful but I think currently parquet 2.10 is not released, so user might not able to write this...

**Comment by letian-jiang** [comment]
🤔 I think this doesn't harm as long as old readers can read new files (w. bloom filter length) correctly.

**Comment by tustvold** [comment]
>  as long as old readers can read new files (w. bloom filter length) correctly.

Thrift supports this form of forwards compatibility natively

**Pull request merged**